src/sentry/grouping/__init__.py

```diff
@@ -1 +1,154 @@
+picks one of the many versions of the algorithm.
+
+The default version of the algorithm is selected on a per project basis.
+
+Variants
+--------
+
+`ChecksumVariant`:
+    This is the legacy variant of the grouping algorithm.  Clients at one point
+    in the past were able to provide a grouping hash (the checksum) which was
+    used for grouping exclusively.
+
+`FallbackVariant`:
+    This variant produces always the same hash.  It's used if nothing else works.
+
+`ComponentVariant`:
+    This is the main variant which uses the underlying component based grouping
+    strategy system.  It produces hashes but can also expose the component
+    tree with annotations so users can debug it.
+
+`CustomFingerprintVariant`:
+    This variant is similar to the checksum variant but instead produces
+    fingerprint values which are then hashed.  This can be used both for
+    events produced by the server as well as events which are fingerprinted
+    by the SDK.  If the special `{{ default }}` value is used then this
+    variant is not used.
+
+`SaltedComponentVariant`:
+    This variant is used when the server or client produce a fingerprint
+    that refers with the special `{{ default }}` value to the default
+    grouping algorithm and the default grouping algorithm uses the
+    component based grouping system.
+
+Component Based Grouping
+------------------------
+
+The component based grouping system produces a tree of nodes (referred to
+as component) to produce the hash.  These are defined in
+`sentry.grouping.component`.  A component has an `id` which is a string
+identifying the component in the tree.  This `id` is not unique in the tree
+but unique on on one level of the hierarchy.  It has an optional `hint` which is
+a string that carries information for the user about why a value was used, not
+used or why it might have been modified.  Additionally it can carry `values`
+which are components again or primitives (strings or integers).  Lastly a
+component has a boolean `contributes` flag.  This flag when set to `False`
+removes a component (and it's children) entirely from the grouping output.
+
+Here an example of how components can be used::
+
+    function_name = 'lambda$1234'
+    threads = GroupingComponent(
+        id="function",
+        values=[function_name],
+        contributes=False,
+        hint='Unused because generated code'
+    )
+
+Strategies and Strategy Configurations
+--------------------------------------
+
+Component based grouping is triggered by strategies.  Strategies define the
+logic for how to generate a component for an interface in the event.  Each
+strategy can only produce a component for one interface.  For instance a
+stacktrace strategy can produce a component tree for a stacktrace.  Because
+events can have different forms and different strategies for the same interface
+exist, strategy configurations define which ones are picked.
```

Comment:
```suggestion
strategy configurations define which ones are picked.
```

This word made the sentence harder for me to understand.